### DISCUSSION TITLE
Print array using the map function

### DISCUSSION THREAD

**Discussion by trya2l** [typst/typst#3472] [Q&A] [answered]
Hi,

I was playing a bit while learning to use Typst, and following the last tutorial on templates, I tried to print the content of an array using the `map` function.

Apparently this is not possible, and I should use a `for` loop.

Is it really impossible to print an array with something like this ?

`#authors.map(author => [#author love]))`

Thanks :smiley:

**Comment by andreasKroepelin** [comment, marked as answer]
`#authors.map(author => [#author love]))` just creates a new array where each element is some content. To join those contents into one content element, use `join`:

```typ
#let authors = ("Einstein", "Newton", "Darwin")
#authors.map(author => [#author was great. ]).join()
```
![rendered.png](https://github.com/typst/typst/assets/42342396/1208ca58-22f4-4a11-a8ad-e84a11a4214c)

**Reply by trya2l** [reply]
Ok thanks, this is great :)
